subsys/storage/flash_map/flash_map.c

```diff
@@ -141,12 +140,11 @@ flash_page_cb cb, struct layout_data *cb_data)
 	cb_data->ret_len = *cnt;
 	cb_data->status = 0;
 
-	flash_dev = device_get_binding(fa->fa_dev_name);
-	if (flash_dev == NULL) {
+	CHECKIF(fa->fa_dev == NULL) {
```

Comment:
Yes. I might be wrong here: the idea is that since the flash_area object is generated at compile time, the `fa_dev` is expected to be attached or compilation error to occur. So there will always be some `fa_dev` , it might not be the right one if DST or custom flash map is not OK, but it will be there.